File: Python/Lamp/Lamp.py
class Lamp:
    __isOn = None
       
    def __init__(self,*args):
        if len(args)==0:
            self.__isOn = False
        else:
            self.__isOn= args[0]
            
    def turnOn(self):
        self.__isOn = True
    
    def turnOff(self):
        self.__isOn = False   
        
        
    def flip(self):
        self.__isOn = not self.__isOn
    
    def isOn(self):
        return self.__isOn
        
    def setLamp(self,isOn):
        self.__isOn = isOn
          
    def __str__(self):
        if self.__isOn:
            out = "{:3s}".format("ON")
        else:
            out = "{:3s}".format("OFF")
        return out
    
    def equals(self,other):
        return self.__isOn == other.__ison

File: Java/Lamp/src/Lamp.java
public class Lamp {

    private boolean isOn;

    public Lamp() {
        this.isOn = false;
    }

    public Lamp(boolean isOn) {
        this.isOn = isOn;
    }

    public void turnOn() {
        this.isOn = true;
    }

    public void turnOff() {
        this.isOn = false;
    }

    public void flip() {
        this.isOn = !this.isOn;
    }

    public boolean isOn() {
        return this.isOn;
    }

    public void setLamp(boolean isOn) {
        this.isOn = isOn;
    }

    @Override
    public String toString() {
        String out;
        if (this.isOn)
            out = String.format("%3s", "ON");
        else
            out = String.format("%3s", "OFF");
        return out;
    }

    @Override
    public boolean equals(Object obj) {
        Lamp other = (Lamp) obj;
        if (isOn != other.isOn)
            return false;
        return true;
    }

}
